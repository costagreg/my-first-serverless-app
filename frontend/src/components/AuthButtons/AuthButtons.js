import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import './AuthButtons.scss'

const AuthButtons = ({ isLogin, setIsLogin }) => (
  <div className="authButtons">
    <button
      className={classNames('authButtons__login', {
        'authButtons__login--active': isLogin,
      })}
      onClick={() => setIsLogin(true)}
    >
      Login
    </button>
    <button
      className={classNames('authButtons__signup', {
        'authButtons__signup--active': !isLogin,
      })}
      onClick={() => setIsLogin(false)}
    >
      Signup
    </button>
  </div>
)

AuthButtons.propTypes = {
  setIsLogin: PropTypes.func,
}

export default AuthButtons
