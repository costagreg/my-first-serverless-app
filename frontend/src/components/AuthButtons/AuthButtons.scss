.authButtons {
  display: flex;
  justify-content: center;
  &__login, &__signup{
    width: 100px;
    height: 100px;
    background: white;
    border: 1px solid #cecece;
    border-radius: 10px;
    margin: 0 5px;
  }
}