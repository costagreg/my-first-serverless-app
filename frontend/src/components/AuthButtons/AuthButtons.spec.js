import React from 'react'
import AuthButtons from './AuthButtons'
import { render } from '@testing-library/react'

describe('AuthButtons', () => {
  describe('@renders', () => {
    it('renders <AuthButtons /> without any error', () => {
      const { asFragment } = render(<AuthButtons />)

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
