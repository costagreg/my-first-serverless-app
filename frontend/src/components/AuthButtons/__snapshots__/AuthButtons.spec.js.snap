// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthButtons @renders renders <AuthButtons /> without any error 1`] = `
<DocumentFragment>
  <div
    class="authButtons"
  >
    <button
      class="authButtons__login"
    >
      Login
    </button>
    <button
      class="authButtons__signup authButtons__signup--active"
    >
      Signup
    </button>
  </div>
</DocumentFragment>
`;
